homeassistant/components/anthemav/config_flow.py

```diff
@@ -0,0 +1,123 @@
+"""Config flow for Anthem A/V Receivers integration."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+import anthemav
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME, CONF_PORT
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import (
+    CONF_MODEL,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DOMAIN,
+    EMPTY_MAC,
+    UNKNOWN_MODEL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Anthem A/V Receivers."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        deviceinfo_received = asyncio.Event()
+
+        @callback
+        def async_anthemav_update_callback(message):
+            """Receive notification from transport that new data exists."""
+            _LOGGER.debug("Received update callback from AVR: %s", message)
+            if "IDN" in message:
+                _LOGGER.debug("received macaddress")
+                user_input[CONF_MAC] = None
+            elif "IDM" in message:
+                _LOGGER.debug("received Model")
+                user_input[CONF_MODEL] = None
+            if CONF_MAC in user_input and CONF_MODEL in user_input:
+                deviceinfo_received.set()
+
+        avr = None
+        try:
```

Comment:
Reduce the scope of this try/except as much as possible, and also catching Exception is too broad.  Perhaps the client library needs to throw its own specific exceptions instead.

Perhaps the `CannotReceiveDeviceInfo` case should be pulled out of the try block since it can just add to the error dict directly rather than using exceptions.

Reply:
Did you consider setting CONF_MAC and CONF_MODEL in the callback above? It seems like it could be simpler to just fill in the values there rather than needing to check them again below.

Reply:
Not sure how I would do this. The callback only return the raw data, and I need to get the actual value from the object, that is not part of the callback.
I'll separate the try/except in 2 (1 for the reconnect, 1 for the wait of the 2 commands)